pkg/routers/api.go

```diff
@@ -63,35 +63,37 @@ func forbiddenHandler(message string) http.HandlerFunc {
 }
 
 // NewDefaultAPIRouter creates a new router with the default backing optimizely.Cache
-func NewDefaultAPIRouter(optlyCache optimizely.Cache, agentConf config.AgentConfig, metricsRegistry *metrics.Registry) http.Handler {
-	conf := agentConf.API
-	authProvider := middleware.NewAuth(&conf.Auth)
+func NewDefaultAPIRouter(optlyCache optimizely.Cache, conf config.AgentConfig, metricsRegistry *metrics.Registry) http.Handler {
+	authProvider := middleware.NewAuth(&conf.API.Auth)
 	if authProvider == nil {
 		log.Error().Msg("unable to initialize api auth middleware.")
 		return nil
 	}
 
-	authHandler := handlers.NewOAuthHandler(&conf.Auth)
+	authHandler := handlers.NewOAuthHandler(&conf.API.Auth)
 	if authHandler == nil {
 		log.Error().Msg("unable to initialize api auth handler.")
 		return nil
 	}
 
 	overrideHandler := handlers.Override
-	if !conf.EnableOverrides {
+	if !conf.API.EnableOverrides {
 		overrideHandler = forbiddenHandler("Overrides not enabled")
 	}
 
-	nStreamHandler := handlers.NotificationEventSteamHandler
-	if !conf.EnableNotifications {
-		nStreamHandler = forbiddenHandler("Notification stream not enabled")
+	nStreamHandler := forbiddenHandler("Notification stream not enabled")
+	if conf.API.EnableNotifications {
+		nStreamHandler = handlers.NotificationEventStreamHandler(handlers.DefaultNotificationReceiver)
+		if conf.Synchronization.Notification.Enable {
```

Comment:
Do we need to enable both to turn on notification stream? What about sync.notification.enable overrides?

Reply:
Yes, top level API.EnableNotifications means Notification API is enabled/disabled. then if it is enabled, then there are two choices. By default, synchronization is disabled.

If the config is changed or overridden, the server needs to be restarted to get the effect.